Project: 03_flask-api
Commit: 9c998897f67d8aa959dc3005d7d22f36568b6938
File: flask_api/parsers.py

============================================================
diff --git a/flask_api/parsers.py b/flask_api/parsers.py
index 56f45f6..d056210 100644
--- a/flask_api/parsers.py
+++ b/flask_api/parsers.py
@@ -2,8 +2,8 @@ import json
 
 from werkzeug.formparser import MultiPartParser as WerkzeugMultiPartParser
 from werkzeug.formparser import default_stream_factory
-from werkzeug.urls import url_decode_stream
 
+from flask_api.helpers import url_decode_stream
 from flask_api import exceptions
 
 
