Project: 05_Pillow
Commit: b3d1836907796e6d6f3c590c9a1860f6ecb04246
File: winbuild/build_prepare.py

============================================================
diff --git a/winbuild/build_prepare.py b/winbuild/build_prepare.py
index 76f05bd..186a80c 100644
--- a/winbuild/build_prepare.py
+++ b/winbuild/build_prepare.py
@@ -116,7 +116,7 @@ V = {
     "BROTLI": "1.1.0",
     "FREETYPE": "2.14.1",
     "FRIBIDI": "1.0.16",
-    "HARFBUZZ": "11.5.0",
+    "HARFBUZZ": "12.1.0",
     "JPEGTURBO": "3.1.2",
     "LCMS2": "2.17",
     "LIBAVIF": "1.3.0",
