Project: 10_click
Commit: 7c7ec36354f49d1a092cb077fa4881ea4d70ba01
File: tests/test_context.py

============================================================
diff --git a/tests/test_context.py b/tests/test_context.py
index 537a2a4..816154d 100644
--- a/tests/test_context.py
+++ b/tests/test_context.py
@@ -478,7 +478,41 @@ def test_with_resource_exception() -> None:
             )
             raise TestException()
 
+
+def test_with_resource_nested_exception() -> None:
+    class TestContext(AbstractContextManager[list[int]]):
+        _handle_exception: bool
+        _base_val: int
+        val: list[int]
+
+        def __init__(self, base_val: int = 1, *, handle_exception: bool = True) -> None:
+            self._handle_exception = handle_exception
+            self._base_val = base_val
+
+        def __enter__(self) -> list[int]:
+            self.val = [self._base_val]
+            return self.val
+
+        def __exit__(
+            self,
+            exc_type: type[BaseException] | None,
+            exc_value: BaseException | None,
+            traceback: TracebackType | None,
+        ) -> bool | None:
+            if not exc_type:
+                self.val[0] = self._base_val - 1
+                return None
+
+            self.val[0] = self._base_val + 1
+            return self._handle_exception
+
+    class TestException(Exception):
+        pass
+
+    ctx = click.Context(click.Command("test"))
+    base_val = 1
     base_val_nested = 11
+
     with ctx.scope():
         rv = ctx.with_resource(TestContext(base_val=base_val))
         rv_nested = ctx.with_resource(TestContext(base_val=base_val_nested))
